pytensor/link/mlx/dispatch/core.py

```diff
@@ -0,0 +1,279 @@
+                warnings.warn(
+                    "MLX does not support float64 on GPU. Automatically casting to float32. "
+                    "This may result in reduced precision. To avoid this warning, "
+                    "explicitly use float32 in your code or set floatX='float32' in PyTensor config.",
+                    UserWarning,
+                    stacklevel=3,
+                )
+                return mx.float32
+            else:
+                return mx.float64
+        elif dtype_str == "bfloat16":
+            return mx.bfloat16
+        elif dtype_str == "complex64":
+            return mx.complex64
+        elif dtype_str == "complex128":
+            if auto_cast_unsupported:
+                warnings.warn(
+                    "MLX does not support complex128. Automatically casting to complex64. "
+                    "This may result in reduced precision. To avoid this warning, "
+                    "explicitly use complex64 in your code.",
+                    UserWarning,
+                    stacklevel=3,
+                )
+                return mx.complex64
+            else:
+                # Return the original even though it might fail
+                # This allows users to opt out of auto-casting if needed
+                return mx.complex64  # MLX doesn't have complex128, so fallback
+    # Return as is if it's already an MLX dtype or not a recognized string
+    return dtype_str
+
+
+@mlx_funcify.register(MakeVector)
+def mlx_funcify_MakeVector(op, **kwargs):
+    dtype = convert_dtype_to_mlx(op.dtype)
+
+    def makevector(*x):
+        return mx.array(x, dtype=dtype)
+
+    return makevector
+
+
+@mlx_funcify.register(TensorFromScalar)
+def mlx_funcify_TensorFromScalar(op, **kwargs):
+    def tensor_from_scalar(x):
+        return x  # already an MLX array / scalar
+
+    return tensor_from_scalar
+
+
+@mlx_funcify.register(ScalarFromTensor)
+def mlx_funcify_ScalarFromTensor(op, **kwargs):
+    def scalar_from_tensor(x):
+        "We can't not return a scalar in MLX without trigger evaluation"
+        return x
+
+    return scalar_from_tensor
+
+
+@mlx_funcify.register(Tri)
+def mlx_funcify_Tri(op, node, **kwargs):
+    # node.inputs  ->  N, M, k
+    const_args = [getattr(inp, "data", None) for inp in node.inputs]
+    dtype = convert_dtype_to_mlx(op.dtype)
+
+    def tri(*args):
+        # Replace args with compile-time constants when available
+        args = [
+            arg if const_a is None else const_a
+            for arg, const_a in zip(args, const_args, strict=True)
+        ]
+        return mx.tri(*args, dtype=dtype)
+
+    return tri
+
+
+@mlx_funcify.register(AllocEmpty)
+def mlx_funcify_AllocEmpty(op, **kwargs):
+    dtype = convert_dtype_to_mlx(op.dtype)
+
+    def allocempty(*shape):
+        return mx.zeros(shape, dtype=dtype)
+
+    return allocempty
+
+
+@mlx_funcify.register(Alloc)
+def mlx_funcify_Alloc(op, node, **kwargs):
+    def alloc(x, *shape):
```

Comment:
We have a more clean implementation in JAX we check if the shape is static outside of the returned function. Check it. Same with reshape and other shape ops that take integers as arguments